### DISCUSSION TITLE
How to run job if tag is v*

### DISCUSSION THREAD

**Discussion by mahnunchik** [community/community#25911] [Actions]
<p>How to run job if tag is versioned, for example v1.2.3?</p>

**Comment by airtower-luna** [comment]
<p>You can use an <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags"><code>on.push.tags</code> filter</a> in the workflow, probably something similar to this:</p>
<pre><code class="lang-yaml">on:
  push:
    tags:
      - 'v*'
</code></pre>

**Comment by mahnunchik** [comment]
<p>Thank you,</p>
<p>This solution is for whole workflow. The question is how to make condition by tag for specific <strong>job</strong> of workflow.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Oh, right! In that case you could try to build an <code>if</code> condition for the job that checks <code>github.event.ref</code>, maybe something like this:</p>
<pre><code class="lang-yaml">if: startsWith(github.event.ref, 'refs/tags/v')
</code></pre>
<p>That’s assuming the job will only be called for push events, otherwise you’d have to check the event type first.</p>

**Comment by mahnunchik** [comment]
<p>It seems it is solution. Thank you!</p>
<p>I’ve tested with <code>github.ref</code></p>
<p>There is missing documentation what the ref means <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context</a></p>

**Comment by airtower-luna** [comment]
<p>Yeah, that could use a link to something like the <a href="https://git-scm.com/book/en/v2/Git-Internals-Git-References">Git Internals - Git References</a> chapter of the Git Book. Also if you want to know which <code>refs</code> are available in your local repository try <a href="https://git-scm.com/docs/git-show-ref"><code>git show-ref</code></a>.</p>
